animation_nodes/nodes/spline/splines_from_edges_utils.pyx

```diff
@@ -0,0 +1,111 @@
+from ... data_structures cimport (
+    Mesh, Vector3DList, EdgeIndicesList, PolySpline,
+    VirtualDoubleList, FloatList, IntegerList
+)
+
+def splinesFromBranches(Vector3DList vertices, EdgeIndicesList edges, VirtualDoubleList radii):
+    cdef int i, j
+    cdef int edgesAmount = edges.length
+    cdef int verticesAmount = vertices.length
+
+    # Compute how many neighbour each vertex have.
+    cdef IntegerList neighboursAmounts = IntegerList.fromValue(0, length = verticesAmount)
+    for i in range(edgesAmount):
+        neighboursAmounts[edges.data[i].v1] += 1
```

Comment:
for performance reasons you should use `.data` when accessing list elements from cython code.

Reply:
Was just going to ask you about this. Will do.